### PR TITLE
ddata replicator stops but doesn't look like it can be restarted easily #5129

### PR DESCRIPTION
**Pull request by andyfurnival**
Wrapped DistributedData Replicator in a Backoff supervisor, so temporary durable store failures don't require a full system restart to recover.

I didn't add this directly into Replicator.Props as this changed the behaviour of many of the multi-node tests, and the trade-off of reworking tests for supporting a supervisor didn't make a whole lot of sense

I had considered putting the supervisor directly on the LmdbDurableStore, however we still run the risk of losing the Replicator with no means of recover (even when durable store isn't used)

Fixed a slightly annoying missing end bracket in ddata logs for GSet

### CODE REVIEW

File: src/contrib/cluster/Akka.DistributedData/DistributedData.cs [hunk 1]
```diff
@@ -83,10 +85,23 @@ public DistributedData(ExtendedActorSystem system)
             else
             {
                 var name = config.GetString("name", null);
-                Replicator = system.ActorOf(Akka.DistributedData.Replicator.Props(_settings), name);
+                Replicator = system.ActorOf(GetSupervisedReplicator(_settings, name), name+"Supervisor");
             }
         }
 
+        private static Props GetSupervisedReplicator(ReplicatorSettings settings, string name) => BackoffSupervisor.Props(
```
Comment (Aaronontheweb) [inline src/contrib/cluster/Akka.DistributedData/DistributedData.cs:92]: The impact this has on the actor path should not affect replication - HOWEVER, when introducing Akka.NET v1.4.22 into a cluster that is running v1.4.21, the replication system will break temporarily due to:

https://github.com/akkadotnet/akka.net/blob/dev/src/contrib/cluster/Akka.DistributedData/Replicator.cs#L1104

Comment (Aaronontheweb) [reply src/contrib/cluster/Akka.DistributedData/DistributedData.cs:92]: We should introduce a HOCON config setting:

```
akka.cluster.distributed-data.recreate-on-failure = off
```

When that setting is off, we don't use the `BackoffSupervisor` - and we will keep that setting to `off` by default. That way this change can be introduced in a non-breaking fashion. When you want to use the `BackoffSupervisor`, set this setting to `on`.
```

**Review by Aaronontheweb — APPROVED** [review]
See my comment about breaking changes during upgrades

File: src/contrib/cluster/Akka.DistributedData/DistributedData.cs [hunk 2]
```diff
@@ -83,10 +85,23 @@ public DistributedData(ExtendedActorSystem system)
             else
             {
                 var name = config.GetString("name", null);
-                Replicator = system.ActorOf(Akka.DistributedData.Replicator.Props(_settings), name);
+                Replicator = system.ActorOf(GetSupervisedReplicator(_settings, name), name+"Supervisor");
```
[the remark this answers is not in the document]
Comment (Arkatufus) [reply src/contrib/cluster/Akka.DistributedData/DistributedData.cs:88]: Just to be clear, the new HOCON flag should switch between these 2 implementation, depending on its value, correct?

[the remark this answers is not in the document]
Comment (Aaronontheweb) [reply src/contrib/cluster/Akka.DistributedData/DistributedData.cs:88]: Correct

**Comment by Aaronontheweb** [comment]
I think the tests introduced here might be a little racy - probably need to allow more time for the first seed node to self-join

File: src/contrib/cluster/Akka.DistributedData/DistributedData.cs (hunk 1 above)
Comment (andyfurnival) [reply src/contrib/cluster/Akka.DistributedData/DistributedData.cs:92]: I've done that change, to default to the original implementation, with a setting to enable backoff.

**Pull request merged**